projects/telemetry/src/nmea_checksum.c

```diff
@@ -0,0 +1,82 @@
+#include "nmea_checksum.h"
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define NMEA_CHECKSUM_HEX_ERROR 16
```

Comment:
Should probably add a comment mentioning why this is valid as an error. Since you're dealing with hex it might be easier to understand to use `0x[y]0` where `y` is a valid hex digit.